cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -618,13 +618,13 @@ private Path getProperInstallationSubDirOf(Path path, Path archiveFile) {
   @Override
   public void extractZip(Path file, Path targetDir) {
 
-    extractArchive(file, targetDir, in -> new ZipArchiveInputStream(in));
+    extractZipArchive(file, targetDir);
   }
 
   @Override
   public void extractTar(Path file, Path targetDir, TarCompression compression) {
 
-    extractArchive(file, targetDir, in -> new TarArchiveInputStream(compression.unpack(in)));
+    extractTarArchive(file, targetDir, in -> new TarArchiveInputStream(compression.unpack(in)));
```

Comment:
Couldn't the TarArchiveInputStream be moved to the extractTarArchive method too?

Reply:
@jan-vcapgemini I am resolving this now without waiting for you since this code is not contained in this PR anymore.